lux/vislib/altair/Histogram.py

```diff
@@ -45,48 +44,52 @@ def initialize_chart(self):
 
         x_min = self.vis.min_max[msr_attr.attribute][0]
         x_max = self.vis.min_max[msr_attr.attribute][1]
+        x_range = abs(x_max - x_min)
 
         if isinstance(msr_attr.attribute, str):
             msr_attr.attribute = msr_attr.attribute.replace(".", "")
 
         colval = self.vis.data[msr_attr.attribute]
-        x_range = abs(max(colval) - min(colval))
-        plot_range = abs(x_max - x_min)
-        markbar = x_range / plot_range * 12
+        markbar = get_bin_size(self.data[msr_attr.attribute])
+        # Default when IQR was too small
+        if markbar < (x_range / 24):
+            markbar = (x_max - x_min) / 12
 
         self.data = AltairChart.sanitize_dataframe(self.data)
+        end_attr_abv = str(msr_attr.attribute) + "_end"
+        self.data[end_attr_abv] = self.data[str(msr_attr.attribute)].apply(lambda x: x + markbar)
```

Comment:
If the data is a data frame, you could just write column + mark around and it would be vectorized (and faster).

Reply:
Thank you for this tip!